### PR TITLE
[rosetta] Rosetta API docs

### PR DESCRIPTION
**Pull request by gregnazario**
### Description
An initial stab at talks about the implementation and details about rosetta.

This is mostly just for internal operational purposes, and not for a user guide or anything specific around Rosetta.  Rosetta is still use at your own risk at this point.

### Test Plan
None, it's documentation!

- - -
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/4807)

### CODE REVIEW

**Review by rajkaramchedu — APPROVED** [review]
Since this won't go on aptos.dev, I'd say, "ship it!"

File: crates/aptos-rosetta-cli/README.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+## Rosetta CLI
+
+The Rosetta CLI is used for testing, and is not used for any production use case.  The CLI uses the same client as the end to end tests to ensure consistency, but with the caveat that it will panicc for cases that aren't expected.
+
```
Comment (rajkaramchedu) [inline crates/aptos-rosetta-cli/README.md:4]: panicc → panic

File: crates/aptos-rosetta/README.md [hunk 2]
```diff
@@ -3,12 +3,95 @@ This implementation is built for running a local proxy against
 a local full node.  However, for testing purposes, this can be used
 against an external REST endpoint.
 
+## Architecture
+
+Rosetta works as a sidecar to an Aptos full node.  Rosetta then proxies the Rosetta standard
```
Comment (rajkaramchedu) [inline crates/aptos-rosetta/README.md:8]: full node → fullnode

**Review by rajkaramchedu — APPROVED** [review]

File: crates/aptos-rosetta-cli/README.md (hunk 1 above)
Comment (clay-aptos) [reply crates/aptos-rosetta-cli/README.md:4]: I fixed this while making some other edits in a commit here.

**Review by clay-aptos — APPROVED** [review]
Thanks, Greg. This LGTM with some edits I made directly in a couple of commits here.

**Pull request merged**